sdk/dapp-kit/src/components/WalletProvider.tsx

```diff
@@ -93,6 +95,28 @@ export function WalletProvider({
 
 	useUnsafeBurnerWallet(enableUnsafeBurner);
 
+	useEffect(() => {
```

Comment:
If you wanna make me really happy, you can pull this into a `useAutoconnect` hook just to keep the logic a little more separate / isolated.